evap/student/views.py

```diff
@@ -24,13 +26,16 @@
 
 @participant_required
 def index(request):
-    # retrieve all courses which have evaluations that are not in state "new" and in which the user participates
-    courses = Course.objects.filter(
-        evaluations__participants=request.user,
-        evaluations__state__in=['prepared', 'editor_approved', 'approved', 'in_evaluation', 'evaluated', 'reviewed', 'published']
-    ).distinct().prefetch_related('semester', 'grade_documents', 'type', 'evaluations', 'evaluations__participants', 'evaluations__voters')
-    # retrieve all evaluations which the user can see that are not new
-    evaluations = [evaluation for course in courses for evaluation in course.evaluations.all() if evaluation.can_be_seen_by(request.user)]
+    query = (Evaluation.objects
+        .filter(course__evaluations__participants=request.user)
+        .exclude(state="new")
```

Comment:
Here's an edge case: Course C1 has evaluations E1 (new, user does participate) and E2 (not new, user does not participate). Right now the code will include E2 for no reason apparent to the user, because the filter clause looks for any participation, and not only for participations of non-new evaluations.

Reply:
added additional filter, thanks!